Dockerfile

```diff
@@ -0,0 +1,63 @@
+FROM debian:stretch
+
+# Install required system packages
+RUN apt-get update && apt-get install -y \
+    apt-utils \
+    automake \
+    g++ \
+    make \
+    libtool \
+    pkg-config \
+    doxygen \
+    libdb++-dev \
+    curl \
+    bsdmainutils \
+    libboost-all-dev \
+    libssl-dev \
+    libevent-dev
+
+# Install Berkeley DB 4.8
+RUN curl -L http://download.oracle.com/berkeley-db/db-4.8.30.tar.gz | tar -xz -C /tmp && \
+    cd /tmp/db-4.8.30/build_unix && \
+    ../dist/configure --enable-cxx --includedir=/usr/include/bdb4.8 --libdir=/usr/lib && \
+    make -j$(nproc) && make install && \
+    cd / && rm -rf /tmp/db-4.8.30
+
+RUN useradd -m -U xsnd
+
+COPY . /tmp/xsncore/
+
+RUN cd /tmp/xsncore && \
+    ./autogen.sh && \
+    ./configure --without-gui --prefix=/usr && \
+    make -j$(nproc) && \
+    make check && \
+    make install && \
+    cd / && rm -rf /tmp/xsncore
+
+# Remove unused packages
+RUN apt-get remove -y \
+    automake \
+    g++ \
+    make \
+    libtool \
+    pkg-config \
+    doxygen \
+    libdb++-dev \
+    curl \
+    bsdmainutils \
+    libboost-all-dev \
+    libssl-dev \
+    libevent-dev
+
+USER xsnd
+
+RUN mkdir /home/xsnd/.xsncore
```

Comment:
from https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user

> WORKDIR
>Dockerfile reference for the WORKDIR instruction
>
>For clarity and reliability, you should always use absolute paths for your WORKDIR. Also, you >should use WORKDIR instead of proliferating instructions like RUN cd … && do-something, >which are hard to read, troubleshoot, and maintain.

Relative Path's won't work after setting the user since it doesn't change the current workdir.

Reply:
we could use WORKDIR which would create the directory but without respecting the USER directive. So it doesn't really make sense for us. Therefore to say relative path's won't work since the USER directive won't change the workdir.

Reply:
Got it, my concern is that at the moment, updating the username isn't that simple because it is in several places, as I said before, I would name it `xsn`.

Reply:
okay changed the name